File: packages/cloud-adapters/observability/interface.ts
// Observability adapter axis — one push surface per signal (logs /
// metrics / traces). Bare-metal pushes to Loki / Mimir / Tempo over
// OTLP+HTTP-push; cloud impls map to CloudWatch+X-Ray / Cloud Logging+
// Trace / Azure Monitor+App Insights.

export interface LogRecord {
  readonly timestamp: number;
  readonly severity: string;
  readonly body: string;
  readonly labels: Record<string, string>;
}

export interface MetricPoint {
  readonly name: string;
  readonly value: number;
  readonly timestamp: number;
  readonly labels: Record<string, string>;
}

export interface SpanRecord {
  readonly traceId: string;
  readonly spanId: string;
  readonly name: string;
  readonly startTimeNs: number;
  readonly endTimeNs: number;
  readonly attributes: Record<string, string>;
}

export interface ObservabilityAdapter {
  pushLogs(records: LogRecord[]): Promise<void>;
  pushMetrics(points: MetricPoint[]): Promise<void>;
  pushTraces(spans: SpanRecord[]): Promise<void>;
}

File: packages/cloud-adapters/observability/interface.go
// Package observability defines the cloud-adapter observability axis interface.
// Bare-metal impl: Loki + Mimir + Tempo. Cloud impls: CloudWatch+X-Ray /
// Cloud Logging+Trace / Azure Monitor+App Insights.
package observability

import "context"

type LogRecord struct {
	Timestamp int64
	Severity  string
	Body      string
	Labels    map[string]string
}

type MetricPoint struct {
	Name      string
	Value     float64
	Timestamp int64
	Labels    map[string]string
}

type SpanRecord struct {
	TraceID     string
	SpanID      string
	Name        string
	StartTimeNs int64
	EndTimeNs   int64
	Attributes  map[string]string
}

type Adapter interface {
	PushLogs(ctx context.Context, records []LogRecord) error
	PushMetrics(ctx context.Context, points []MetricPoint) error
	PushTraces(ctx context.Context, spans []SpanRecord) error
}
